packages/jira-adapter/src/filters/sla_addition_deployment.ts

```diff
@@ -0,0 +1,177 @@
+    return []
+  }
+}
+
+const updateDefaultSla = async (
+  change: AdditionChange<InstanceElement>,
+  client: JiraClient,
+  serviceId: number,
+  jsmApiDefinitions: configDeprecated.AdapterDuckTypeApiConfig,
+): Promise<void> => {
+  change.data.after.value.id = serviceId
+  const emptySlaInstance = change.data.after.clone()
+  emptySlaInstance.value = {}
+  const modifyChange = toChange({ before: emptySlaInstance, after: change.data.after })
+  await defaultDeployChange({
+    change: modifyChange,
+    client,
+    apiDefinitions: jsmApiDefinitions,
+  })
+}
+
+/*
+ * This filter responsible for deploying slas with default names.
+ * all other sla, will be deployed through the standard JSM deployment.
+ */
+const filter: FilterCreator = ({ config, client }) => ({
+  name: 'slaAdditionFilter',
+  deploy: async changes => {
+    const { jsmApiDefinitions } = config
+    if (!config.fetch.enableJSM || jsmApiDefinitions === undefined) {
+      return {
+        deployResult: { appliedChanges: [], errors: [] },
+        leftoverChanges: changes,
+      }
+    }
+    const [slaAdditionChanges, leftoverChanges] = _.partition(
+      changes,
+      change =>
+        isAdditionChange(change) && isInstanceChange(change) && getChangeData(change).elemID.typeName === SLA_TYPE_NAME,
+    )
+
+    if (slaAdditionChanges.length === 0) {
+      return {
+        deployResult: { appliedChanges: [], errors: [] },
+        leftoverChanges: changes,
+      }
+    }
+
+    const projectToDefaultSlaAdditions = _.groupBy(slaAdditionChanges, change => {
+      try {
+        const parent = getParent(getChangeData(change))
+        return parent.elemID.getFullName()
+      } catch (e) {
+        log.error(`failed to get project name for change ${getChangeData(change).elemID.name} due to an error ${e}`)
+        return ''
+      }
+    })
+    const projectToServiceSlas: Record<string, string[][]> = Object.fromEntries(
+      await Promise.all(
+        Object.entries(projectToDefaultSlaAdditions).map(async ([projectName, slaChanges]) => {
+          const parent = getParent(getChangeData(slaChanges[0]))
+          return [projectName, await getExistingSlaNamesAndIds(parent, client)]
+        }),
+      ),
+    )
+    const typeFixedChanges = slaAdditionChanges.map(change => ({
+      action: change.action,
+      data: _.mapValues(change.data, (instance: InstanceElement) =>
+        replaceInstanceTypeForDeploy({
+          instance,
+          config: jsmApiDefinitions,
+        }),
+      ),
+    })) as Change<InstanceElement>[]
+
+    const deployResult = await deployChanges(
+      typeFixedChanges.filter(isInstanceChange).filter(isAdditionChange),
+      async change => {
+        const serviceId = Object.fromEntries(
+          projectToServiceSlas[getParent(getChangeData(change)).elemID.getFullName()] ?? [],
+        )[change.data.after.value.name]
+        if (serviceId === undefined) {
+          await defaultDeployChange({
```

Comment:
readability nit - 
I think it would be easier to read if you create the change depends on `serviceId` and then just call `defaultDeployChange` once, something like - 
```suggestion
      const updatedChange = serviceId === undefined ? change : createSlaModificationChange(change)
       await defaultDeployChange({
          change: updatedChange,
          client,
          apiDefinitions: jsmApiDefinitions,
       })
```